FirebaseMLModelDownloader/Sources/ModelInfo.swift

```diff
@@ -16,44 +16,58 @@ import Foundation
 import FirebaseCore
 
 /// Model info object with details about pending or downloaded model.
-class ModelInfo: NSObject {
+struct ModelInfo {
   /// Model name.
-  var name: String
+  let name: String
 
   // TODO: revisit UserDefaultsBacked
   /// Download URL for the model file, as returned by server.
-  @UserDefaultsBacked var downloadURL: String
+  let downloadURL: String
 
   /// Hash of the model, as returned by server.
-  @UserDefaultsBacked var modelHash: String
+  let modelHash: String
 
   /// Size of the model, as returned by server.
-  @UserDefaultsBacked var size: Int
+  let size: Int
 
   /// Local path of the model.
-  @UserDefaultsBacked var path: String?
+  var path: String?
 
   /// Initialize model info and create user default keys.
-  init(app: FirebaseApp, name: String, defaults: UserDefaults) {
+  init(name: String, downloadURL: String, modelHash: String, size: Int) {
     self.name = name
+    self.downloadURL = downloadURL
+    self.modelHash = modelHash
+    self.size = size
+  }
+
+  init?(name: String, app: FirebaseApp, defaults: UserDefaults) {
```

Comment:
nit: I think it will be more self-documenting with a name like `init?(readingFromDefaults:, name:, app:)`

Reply:
Yep, done!